packages/react-router/modules/useRouter.js

```diff
@@ -0,0 +1,20 @@
+import React, { useContext } from "react";
+import invariant from "tiny-invariant";
+
+import RouterContext from "./RouterContext";
+import matchPath from "./matchPath";
+
+export default function useRouter(options = {}) {
+  invariant(
```

Comment:
Remove checking in `production`?
```js
if (process.env.NODE_ENV !== 'production') {
  invariant(
    typeof useContext === "function",
    "The useRouter hook requires React 16.7 or greater"
  );
}
```

Reply:
The point of `invariant` is to assert that a condition passes. In this case, if `useContext` is not a function the next line will fail. So I'd rather bail with an `invariant` instead of the user getting a `useContext is not a function` error.

Also, we are already using the `dev-expression` babel plugin, which converts this code to:

```js
if (typeof useContext !== 'function') {
  invariant(false);
}
```

in production. So the error message is not part of our production bundle.

Reply:
The best alternative is only to export a function that has that invariant. I dont understand why it should check every render.

Like

```js
export default process.env.NODE_ENV !== ’production’
  ? invariant.bind(null, typeof useContext !== 'function', ’The useRouter hook requires React 16.7 or greater’)
  : useRouter;
```